popup.js

```diff
@@ -169,15 +174,44 @@ function loginProcess() {
         checkAccount();
         chrome.tabs.getSelected(null, function (tab) {
             tablink = tab.url;
-            if (startsWithHTTP(tablink,"http://findagrave.com") || startsWithHTTP(tablink,"http://forums.findagrave.com")) {
-                tablink = tablink.replace("http://findagrave.com", "http://www.findagrave.com");
-                tablink = tablink.replace("http://forums.findagrave.com", "http://www.findagrave.com");
+
+            console.log(collections);
+            // Select collection
+            for (var i=0; i<collections.length; i++) {
+              if (startsWithHTTP(tablink, collections[i].url)) {
+                collection = collections[i];
+                break;
+              }
+            }
+            if (collection == undefined) {
+              // TODO: display error
+              console.log("Could not find collection");
+            }
+            console.log("collection URL: "+collection.url);
+
+            // prepareUrl
+            if (collection.prepareUrl) {
+              tablink = collection.prepareUrl(tablink);
             }
+
+            // TODO: which collection is that?
             if (startsWithHTTP(tablink,"http://www.ancestry.") && !startsWithHTTP(tablink,"http://www.ancestry.com") && tablink.contains("/genealogy/records/")) {
                 var ancestrystart = tablink.split("www.ancestry.");
                 //Replace domain for other countries, such as http://www.ancestry.ca/genealogy/records/abraham-knowlton_17477348
                 tablink = ancestrystart[0] + "www.ancestry.com" + ancestrystart[1].substring(ancestrystart[1].indexOf("/"));
             }
+
+            // Parse data
+            if (collection.parseData) {
+                console.log("Going to parse data now");
+                collection.parseData();
+            } else {
+                console.log("No parseData function");
+                setMessage("#f9acac", 'SmartCopy does not currently support parsing this page / site / collection.');
+                document.querySelector('#loginspinner').style.display = "none";
+            }
+
+            // TODO: migrate all this to parseData()
             if (startsWithMH(tablink, "research/collection") || startsWithMH(tablink, "research/record") || (startsWithHTTP(tablink,"http://www.findagrave.com") && !tablink.contains("page=gsr")) ||
                 startsWithHTTP(tablink,"http://www.wikitree.com") || startsWithHTTP(tablink,"http://trees.ancestry.") || startsWithHTTP(tablink,"http://person.ancestry.") || startsWithHTTP(tablink,"http://www.werelate.org/wiki/Person") ||
```

Comment:
Where do the `research/collection`, `research/record` and `page=gsr` conditions belong? MH?